openedx/core/djangoapps/oauth_dispatch/docs/decisions/0010-csrf-protection.rst

```diff
@@ -0,0 +1,64 @@
+10. Dealing with Django CSRF Protection in Frontend Apps
+--------------------------------------------------------
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+For background, please see:
+
+* `Transport JWT in HTTP Cookies`_, where we decided the mechanism that frontend apps will use to authenticate with
+  backend API services.
+
+* `Django CSRF Protection`_, which contains a description of Django's CSRF protection.
+
+Frontend apps need a way to obtain a CSRF token when making "unsafe" (POST, PUT, DELETE) requests to backend API services
+implemented as Django applications with CSRF protection enabled. Each backend service is generally deployed with a CSRF
+token cookie scoped to the domain of the given service (ecommerce.edx.org) and named accordingly (ecommerce_csrftoken).
+Each backend service uses a unique SECRET_KEY to produce the CSRF token, so a CSRF token created by one service will not
+be valid for another service. Therefore, frontend apps served by different subdomains will not have access to these CSRF
+token cookies.
+
+.. _Transport JWT in HTTP Cookies: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0009-jwt-in-session-cookie.rst
+.. _Django CSRF Protection: https://docs.djangoproject.com/en/2.1/ref/csrf/
+
+Decisions
+---------
+
+#. **CSRF Token API Endpoint** We will implement an API view that constructs a valid CSRF token and returns
+   it in a JSON response. This API view will be added to the `edx-drf-extensions`_ python library.
+   edx_rest_framework_extensions will be added to INSTALLED_APPS in each backend service as we encounter
+   frontend apps that need to make "unsafe" API requests to a given backend service.
```

Comment:
Why the use of "unsafe" here?  Can it just be `make API requests`?

Reply:
This is shorthand for `POST`, `PUT`, and `DELETE` (Django uses this term in their CSRF documentation. I will make it explicit instead of using the shorthand.